### PR TITLE
Add base path for enable start Fz with specific path

### PR DESCRIPTION
**Pull request by heavenshell**
Add root path option.
```
:Fz ~/.vim/pack/bundle/start
```
vim-fz

### CODE REVIEW

**Comment by mattn** [comment]
@prabirshrestha This can be breaking change of API. How do you think?

**Comment by prabirshrestha** [comment]
I would prefer this if possible: `call fz#run({ args: <q-args>})`. The main reason dict was used is for compatibility. 

In this case rather than args, `basepath` might be better. @heavenshell can you also try with fzf?

**Comment by heavenshell** [comment]
@mattn @prabirshrestha Thank you for reviewed my PR.
Fixed! PTAL!

>@heavenshell can you also try with fzf?

Sorry, what do you mean?

File: autoload/fz.vim [hunk 1]
```diff
@@ -107,12 +107,19 @@ function! fz#run(...)
...
+    let basepath = expand(basepath)
+  endif
+
   " check type
   let typ = get(ctx['options'], 'type', 'cmd')
   if typ == 'cmd'
     let $FZ_IGNORE = get(ctx['options'], 'ignore', '(^|[\/])(\.git|\.hg|\.svn|\.settings|\.gitkeep|target|bin|node_modules|\.idea|^vendor)$|\.(exe|so|dll|png|obj|o|idb|pdb)$')
     let fz_command = get(ctx['options'], 'fz_command', g:fz_command)
     let cmd = get(ctx['options'], 'cmd', g:fz_command_files)
+    let cmd = isdirectory(basepath) ? printf(cmd, basepath) : printf(cmd, '')
```
Comment (mattn) [inline autoload/fz.vim]:
```
let cmd = printf(cmd, basepath)
```

File: autoload/fz.vim [hunk 2]
```diff
@@ -107,12 +107,19 @@ function! fz#run(...)
     return
   endif
 
+  " Get basepath
+  let basepath = get(ctx['options'], 'basepath', '')
+  if basepath != ''
+    let basepath = expand(basepath)
+  endif
```
Comment (mattn) [inline autoload/fz.vim]:
```
let basepath = expand(get(ctx['options'], 'basepath', '.'))
```

Comment (heavenshell) [reply autoload/fz.vim]: 🙆

File: autoload/fz.vim (hunk 1 above)
Comment (heavenshell) [reply autoload/fz.vim]: Fumm, if given path is not valid, nothing to show.

File: autoload/fz.vim (hunk 2 above)
Comment (heavenshell) [reply autoload/fz.vim]: Fixed.

File: autoload/fz.vim (hunk 1 above)
Comment (heavenshell) [reply autoload/fz.vim]: Fixed.

**Comment by prabirshrestha** [comment]
@heavenshell This PR has on how to use `fzf` https://github.com/mattn/vim-fz/pull/9

```
let g:fz_command = 'fzf'
let g:fz_command_files = ''
```

**Comment by heavenshell** [comment]
@prabirshrestha Thx I'll try it later 👍

**Comment by heavenshell** [comment]
@prabirshrestha Is `fzf` working?
I tried following at master branch.
```
let g:fz_command = expand('~/.fzf/bin/fzf')
let g:fz_command_files = ''
```
but show nothing.

I put debug print below https://github.com/mattn/vim-fz/blob/master/autoload/fz.vim#L140

then showed like followings.
```
['/opt/local/bin/zsh', '-c', '/Users/s_ohyanagi/.fzf/bin/fzf -a=ctrl-v,ctrl-x,ct
rl-o,ctrl-t > /var/folders/wk/wtn88t9559j3zpqj_pwxzzh80000gn/T/vE7lP5H/0']
```
and Vim shows `unknown option: -a=ctrl-v,ctrl-x,ctrl-o,ctrl-t`.

**Comment by heavenshell** [comment]
@prabirshrestha Now fzf is working.

vim-fz-fzf

**Comment by heavenshell** [comment]
@prabirshrestha Could you review this?

**Comment by prabirshrestha** [comment]
@heavenshell looks good to me from code perspective although I haven't tried it out yet.

**Comment by mattn** [comment]
Thank you!

**Comment by heavenshell** [comment]
Thank you for review and merge this! ☺️

**Pull request merged**